src/deadline_test_fixtures/deadline/worker.py

```diff
@@ -223,10 +225,60 @@ def start(self) -> None:
         self._start_worker_agent()
 
     def stop(self) -> None:
+        worker_id = self.worker_id
         LOG.info(f"Terminating EC2 instance {self.instance_id}")
         self.ec2_client.terminate_instances(InstanceIds=[self.instance_id])
+
+        """
+        When using autoscaling workers are deleted from fleets, 
+        for non autoscaling fleets we need to manually delete the worker.
+        """
+        if not self.configuration.fleet.autoscaling:
+            self.delete_worker_from_fleet(worker_id)
+
         self.instance_id = None
 
+    def delete_worker_from_fleet(self, worker_id: str):
+        for _ in range(30):
+            response = self.deadline_client.get_worker(
+                farmId=self.configuration.farm_id,
+                fleetId=self.configuration.fleet.id,
+                workerId=worker_id,
+            )
+            if response["status"] == "STOPPED":
+                LOG.info(f"{worker_id} is STOPPED")
+                break
+            time.sleep(5)
+            LOG.info(f"Waiting for {worker_id} to transition to STOPPED status")
+        else:
+            LOG.exception(f"{worker_id} did not transition to a STOPPED status")
+            try:
+                self.force_stopped_status(worker_id)
+            except botocore.exceptions.ClientError as error:
+                LOG.exception(f"Failed to transition {worker_id} to stopped status: {error}")
```

Comment:
It might help to separate this logic into it's own method, for example:

```py
def wait_until_stopped(
    self,
    *,
    max_attempts: int = 30,
    delay_between_attempts: timespan = timespan.seconds(5),
) -> None:
    ...
```

When doing this, I think it would make sense to also de-couple the `force_stopped_status(...)` call. This way the caller has more control, for example:

```py
worker = Worker(...)
try:
    worker.wait_until_stopped(...)
except Worker.Timeout:
    LOG.warning("worker status did not transition to STOPPED, stopping manually...")
    worker.stop()
```

What do you think?

Reply:
Makes sense. Updated in the new revision.